src/user/dto/update-user.dto.ts

```diff
@@ -0,0 +1,4 @@
+import { PartialType } from '@nestjs/mapped-types';
+import { CreateUserDto } from './create-user.dto';
+
+export class UpdateUserDto extends PartialType(CreateUserDto) {}
```

Comment:
Okay ... did you think about using `Patch` here instead of making it all just optional? Still - maybe I'm lacking some type of knowledge about nest!

Reply:
Do you mean Patch as http method?

Reply:
Yup. The update suggest the whole model will be updated always - that is what for it was invented and patch suggest that certain attribute can be changed, not the whole model always